src/devices/Common/Iot/Device/Graphics/BitmapImage.cs

```diff
@@ -70,5 +147,83 @@ public virtual void Clear(Color color = default)
                 }
             }
         }
+
+        /// <summary>
+        /// Sets pixel at specific position
+        /// </summary>
+        /// <param name="x">X coordinate of the pixel</param>
+        /// <param name="y">Y coordinate of the pixel</param>
+        /// <param name="color">Color to set the pixel to</param>
+        /// <remarks>The use of <see cref="SetPixel"/> and <see cref="GetPixel"/> is usually slow. For fast image updates, grab the underlying
+        /// raw buffer by calling <see cref="AsByteSpan"/> or use the <see cref="GetDrawingApi"/> method and use high-level drawing functions.</remarks>
+        public abstract void SetPixel(int x, int y, Color color);
+
+        /// <summary>
+        /// Gets the color of the pixel at the given position
+        /// </summary>
+        /// <param name="x">X coordinate of the pixel</param>
+        /// <param name="y">Y coordinate of the pixel</param>
+        /// <returns>The color of the pixel</returns>
+        /// <remarks>The use of <see cref="SetPixel"/> and <see cref="GetPixel"/> is usually slow. For fast image updates, grab the underlying
+        /// raw buffer by calling <see cref="AsByteSpan"/> or use the <see cref="GetDrawingApi"/> method and use high-level drawing functions.</remarks>
+        public abstract Color GetPixel(int x, int y);
+
+        /// <summary>
+        /// Return the data pointer as a raw span of bytes
+        /// </summary>
+        /// <returns>A span of bytes</returns>
+        public abstract Span<byte> AsByteSpan();
+
+        /// <summary>
+        /// Return the data pointer as a raw span of integers. For 32bit images, this equals to one pixel per entry.
+        /// </summary>
+        /// <returns>A span of integers</returns>
+        public virtual Span<int> AsIntSpan()
```

Comment:
does this need to be virtual? could it be just regular method? do we actually need it? Would be better if callsite did conversion